test/loco/integer/magicseries2.clj

```diff
@@ -0,0 +1,68 @@
+(ns ^{:author "Andrea Richiardi"
+      :doc "Magic Series example #2 + tests. This implementation does not make use of loco's
+            $cardinality, see Magic Series #1 for that.
+            A series S = (S0,...Sn) is magic if Si represents the number of occurrences of i in S.
+            Example: A Magic Series of length 5 is |2|1|2|0|0|"}
+  loco.integer.magicseries2
+  (:require [clojure.test :refer :all]
+            [loco.core :refer :all]
+            [loco.constraints :refer :all]))
+
+(defn series-vars
+  "Returns the control variables of the Magic Series."
+  [series-length]
+  (for [i (range series-length)]
+    [:s i]))
+
+(defn numeric-constraints
+  "Builds the numeric constraints of each Magic Series control variable
+  in input."
+  [series]
+  (let [series-length (count series)]
+    (for [s series] 
+      ($in s 0 (dec series-length)))))
+
+(defn occurrence-count
+  "Given a k element in the Magic Series, returns a var containing the
+  number of its occurrences in the series itself."
+  [i series]
+  (apply $+ (for [s series]
+              ($reify ($= i s)))))
+
+(defn occurrence-constraints
+  "Builds the occurrence constraint of each Magic Series control variable
+  in input."
+  [series]
+  (for [s series]
+    ($= s (occurrence-count (second s) series))))
+
+(defn ms2-model
+  "Return the model."
+  [series-length]
+  (let [series (series-vars series-length)] 
+    (concat series
```

Comment:
Same here... `series` at the top level of the model is simply unnecessary if I'm interpreting this correctly.